src/test/java/com/exasol/adapter/capabilities/CapabilitiesTest.java

```diff
@@ -18,67 +21,116 @@ void beforeEach() {
-                () -> assertEmptyPredicateCapabilities(capabilities), //
-                () -> assertEmptyScalarFunctionCapabilities(capabilities), //
-                () -> assertCapabilitesContainAllOf(capabilities, expectedCapabilities));
+              () -> assertEmptyLiteralCapabilities(capabilities), //
+              () -> assertEmptyPredicateCapabilities(capabilities), //
+              () -> assertEmptyScalarFunctionCapabilities(capabilities), //
+              () -> assertCapabilitesContainAllOf(capabilities, expectedCapabilities));
+    }
+
+    @Test
+    void substractCapabilities() {
+        final MainCapability[] mainCapabilities =
+              {MainCapability.AGGREGATE_GROUP_BY_COLUMN, MainCapability.AGGREGATE_GROUP_BY_EXPRESSION};
+        final LiteralCapability[] literalCapabilities = {LiteralCapability.DATE, LiteralCapability.DOUBLE};
+        final PredicateCapability[] predicateCapabilities = {PredicateCapability.EQUAL, PredicateCapability.BETWEEN};
+        final ScalarFunctionCapability[] scalarFunctionCapabilities =
+              {ScalarFunctionCapability.ADD, ScalarFunctionCapability.ABS};
+        final AggregateFunctionCapability[] aggregateFunctionCapabilities =
+              {AggregateFunctionCapability.APPROXIMATE_COUNT_DISTINCT, AggregateFunctionCapability.AVG};
+        final Capabilities capabilities =
+              this.builder.addMain(mainCapabilities).addLiteral(literalCapabilities).addPredicate(predicateCapabilities)
+                    .addScalarFunction(scalarFunctionCapabilities).addAggregateFunction(aggregateFunctionCapabilities)
+                    .build();
+
+        this.builder = Capabilities.builder();
+        final MainCapability[] mainCapabilitiesToExclude = {MainCapability.AGGREGATE_GROUP_BY_COLUMN};
+        final LiteralCapability[] literalCapabilitiesToExclude = {LiteralCapability.DATE};
+        final PredicateCapability[] predicateCapabilitiesToExclude = {PredicateCapability.EQUAL};
+        final ScalarFunctionCapability[] scalarFunctionCapabilitiesToExclude = {ScalarFunctionCapability.ADD};
+        final AggregateFunctionCapability[] aggregateFunctionCapabilitiesToExclude =
+              {AggregateFunctionCapability.APPROXIMATE_COUNT_DISTINCT};
+        final Capabilities capabilitiesToExclude =
+              this.builder.addMain(mainCapabilitiesToExclude).addLiteral(literalCapabilitiesToExclude)
+                    .addPredicate(predicateCapabilitiesToExclude).addScalarFunction(scalarFunctionCapabilitiesToExclude)
+                    .addAggregateFunction(aggregateFunctionCapabilitiesToExclude).build();
+
+        final Capabilities capabilitiesWithExclusion = capabilities.subtractCapabilities(capabilitiesToExclude);
+        assertAll(() -> assertThat(capabilitiesWithExclusion.getMainCapabilities(),
+              contains(MainCapability.AGGREGATE_GROUP_BY_EXPRESSION)),
+              () -> assertThat(capabilitiesWithExclusion.getMainCapabilities(),
```

Comment:
Superfluous. Since `contains()` tests for exact content match, you don't need to additionally verify that something is missing.

Reply:
fixed